vavr/src/main/java/org/jdbi/v3/core/mapper/VavrTuple2Mapper.java

```diff
@@ -0,0 +1,120 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.mapper;
+
+import io.vavr.Tuple2;
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.statement.StatementContext;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.lang.reflect.TypeVariable;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Optional;
+
+import static org.jdbi.v3.core.generic.GenericTypes.getErasedType;
+import static org.jdbi.v3.core.generic.GenericTypes.resolveType;
+
+/**
+ * core extension to provide functionality similar to {@link org.jdbi.v3.core.mapper.MapEntryMapper}
+ * <p>
+ * TODO refactor common code with core mapper
+ * TODO java9 wont allow usage of package private functions like that
+ *
+ * @param <K>
+ * @param <V>
+ */
+public class VavrTuple2Mapper<K, V> implements RowMapper<Tuple2<K, V>> {
+    private static final TypeVariable<Class<Tuple2>> KEY_PARAM;
+    private static final TypeVariable<Class<Tuple2>> VALUE_PARAM;
+    private static final int TUPLE_VALUE_COLUMN_INDEX = 2;
+
+    static {
+        TypeVariable<Class<Tuple2>>[] mapParams = Tuple2.class.getTypeParameters();
+        KEY_PARAM = mapParams[0];
+        VALUE_PARAM = mapParams[1];
+    }
+
+    private final RowMapper<K> keyMapper;
+    private final RowMapper<V> valueMapper;
+
+    private VavrTuple2Mapper(RowMapper<K> keyMapper, RowMapper<V> valueMapper) {
+        this.keyMapper = keyMapper;
+        this.valueMapper = valueMapper;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static RowMapperFactory factory() {
+        return (type, config) -> {
+            if (type instanceof ParameterizedType && getErasedType(type).equals(Tuple2.class)) {
+                Type keyType = resolveType(KEY_PARAM, type);
+                Type valueType = resolveType(VALUE_PARAM, type);
+
+                RowMapper<?> keyMapper = getKeyMapper(keyType, config);
+                RowMapper<?> valueMapper = getValueMapper(valueType, config);
+
+                return Optional.of(new VavrTuple2Mapper<>(keyMapper, valueMapper));
+            }
+            return Optional.empty();
+        };
+    }
+
+    private static RowMapper<?> getKeyMapper(Type keyType, ConfigRegistry config) {
+        String column = config.get(MapEntryMappers.class).getKeyColumn();
+        if (column == null) {
+            // precedence of col mappers over row mappers to allow for expected behaviour in tuple2 projection
```

Comment:
just noticed, that it might be problematic if you expect a Tuple2 projection of some query but have key and value column (for maps) configured globally! not sure how to resolve this properly

Reply:
What about the interaction with `Configurable#setMapKeyColumn`  and value column? I wanted to be able to reuse the global configuration methods to work with the vavr Map implementation.

Do you see any way this still can be done with the config package as of now?

Reply:
Ah, I see what you mean: you want the convenience of `Configurable.setKeyColumn` / `.setValueColumn`. I think that's reasonable.

In that case I'd probably have your `_1` and `_2` column setters either 1) act as an alias for `MapEntryMappers.keyColumn` and `.valueColumn`, or 2) have them be separate, but use the key/value properties in core as a fallback if the `TupleMappers` properties are unset.

Reply:
I understand and will implement one of those options.

One problem remains: Since my new class would delegate from and to MapEntryMappers I would have to leave it in the same package to access the package private `getXXXColumn()` methods of the delegate

Reply:
You can make those `MapEntryMappers` getters public.